UserProjectRolesTable pagination and search

emeremikwu-lm opened the pull request:
### Description

<!-- REQUIRED
    Provide a concise description of your change. What does it do? Why is it necessary?
    As a guideline, think about how you would describe your change if you were to write a changelog entry for it.
-->

### Addressed Issue

<!-- REQUIRED
    Reference the issue addressed by this PR, e.g. `#1234`.
    Use keywords to signal that this PR resolves the issue,
    causing the issue to be closed automatically when the PR is merged:
        https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

### Additional Details

<!-- OPTIONAL
    If desired, share more technical details about the change here.
    Elaborating on why you implemented the change the way you did can be super helpful to the reviewer.
    Did you consider other solutions? Any problems you ran into along the way?

    Providing screenshots, GIFs or even short clips of the new behavior is a great way to demonstrate
    the change to other community members.
-->

### Checklist

<!-- REQUIRED
    Mark items in this list as done by adding a `x` between the square brackets.
    Non-applicable items may be marked as such by surrounding their text with tildes (`~`).

    This is not meant to be a strict to-do list. If you're unsure about anything,
    just leave it empty for now. The maintainers are happy to assist you in figuring it out!
-->

- [ ] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/hyades/blob/main/CONTRIBUTING.md#pull-requests)
- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://dependencytrack.github.io/hyades/latest/development/documentation/) accordingly

emeremikwu-lm committed:
fix(auth and roles): update deleteRole method to use correct endpoint and emit delete event

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
fix(gitlab integration): simplify button markup in card footer

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
feat(auth and roles): enhance table with pagination and inline search features

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: cleanup

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

ashearin commented on src/i18n/locales/pl.json:
```diff
@@ -964,7 +967,7 @@
     "will_not_fix": "Nie naprawi",
     "workaround_available": "Dostępne obejście",
-    "x_trust_boundary": "Granica zaufania krzyżowego"
+    "x_trust_boundary": "Granica zaufan ia krzyżowego"
```
Was this intentional or  auto change from the extension? Probably best if we leave it alone

emeremikwu-lm committed:
fix: correct typo in Polish translation for trust boundary

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: refactor inline search and pagination for user project roles table

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: comment and unnecessary logic cleanup

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: update table options for pagination and inline search functionality

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: gitlab integration reorganization

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: cleanup css styles and remove unused i18n

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

ashearin commented on src/views/components/UserProjectRolesTable.vue:
```diff
@@ -1,154 +1,318 @@
+import _ from 'lodash';
+
+const defaultTableOptions = {
+  paginationOptions: [5, 10, 25],
```
```suggestion
  paginationOptions: [10, 25, 50],
```
To align with Permissions pagination options

emeremikwu-lm committed:
Update src/views/components/UserProjectRolesTable.vue

Co-authored-by: Allen Shearin <allen.p.shearin@gmail.com>
Signed-off-by: emeremikwu-lm <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: update labels in GitLab integration and enhance CopyField styling

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
refactor: reorganize LDAP, Managed, and OIDC user management views

- Reverted LDAP, Managed, and OIDC vue components back to single file components.
- Created a new UserDetails.vue component to handle user details for LDAP, Managed, and OIDC users.
- Removed userManagementMixin.
- Unified user detail logic and UI, using a userType prop to handle differences between user types.
- Updated API endpoint handling and conditional rendering for managed-user-specific fields.
- Improved maintainability by consolidating similar logic and removing duplication.

Signed-off-by: Emmanuel Meremikwu <emmanuel.meremikwu@lmco.com>

emeremikwu-lm committed:
Update UserDetails.vue

Removed unused function loadUserManagementData

Signed-off-by: emeremikwu-lm <emmanuel.meremikwu@lmco.com>
